### ISSUE TITLE
what is the purpose for subtracting the max logit?

### ISSUE THREAD

**Issue by LiJunnan1992** [HobbitLong/SupContrast#4]
Hi, thanks for the nice code! 

Can I know why do you subtract the max logit? I'm curious what difference does it make to the optimization?

**Comment by HobbitLong** [comment]
Hi, @LiJunnan1992,

We just want to avoid very large logits that lead to NaN problem. It's the same for optimization purpose.
